### PR TITLE
fix: view management, don't show trust & connect when not needed

### PR DESCRIPTION
**Pull request by bastiandoetsch**
### Description

Previously, during plugin startup, "trust & connect" was shown in the welcome dialog. Now, during startup, a message is shown that the plugin is still starting. Views are now initialized with reasonable defaults and only displayed when needed.

Also removed the "feature select" view that was not shown anymore anyway. Boy Scouts Rule!

### Checklist

- [ ] Tests added and all succeed
- [x] Linted
- [x] CHANGELOG.md updated
- [ ] README.md updated, if user-facing

### Screenshots / GIFs
Before:
https://github.com/snyk/vscode-extension/assets/20150761/fdeb9a93-1f4c-4831-95c2-1d0c2eeb8eb2

Screenrecording After:
https://github.com/snyk/vscode-extension/assets/20150761/d2ae2c58-0721-45ac-9690-7083b9ec7d47

### CODE REVIEW

File: src/snyk/common/constants/views.ts [hunk 1]
```diff
@@ -13,12 +13,11 @@ export const SNYK_VIEW_ANALYSIS_IAC = 'snyk.views.analysis.configuration';
 // Having multiple boolean contexts instead of a single context
 // with multiple values helps us to avoid flickering UI.
 export const SNYK_CONTEXT = {
+  NOT_LOADING: 'notLoading', // default to loading state (notLoading = false when boolean is initialized)
```
Comment (cat2608) [inline src/snyk/common/constants/views.ts:16]: I'm new to this project so please dismiss this comment if not relevant 🙏 

I usually find negatives in names for standalone variables a bit confusing to follow. For context/justification about my comment, I'm basing my thinking on articles like [this](https://www.serendipidata.com/posts/naming-guidelines-for-boolean-variables).

I can follow that we use it to display a view or not, right?

```
    // initialize context correctly
    // see package.json when each view is shown, based on context value
    await this.contextService.setContext(SNYK_CONTEXT.NOT_LOADING, false);
```

Does it make sense to call it `showSomething`/`showView`?

Comment (bastiandoetsch) [reply src/snyk/common/constants/views.ts:16]: We need a state where initialization with false is fine :). Else I'm open for any name.

File: src/snyk/snykCode/views/suggestion/codeSuggestionWebviewScript.ts [hunk 2]
```diff
@@ -6,6 +6,9 @@
 /* eslint-disable @typescript-eslint/no-unsafe-call */
 /// <reference lib="dom" />
 
+// eslint-disable-next-line @typescript-eslint/no-unused-vars,@typescript-eslint/no-explicit-any
+declare const acquireVsCodeApi: any;
+
```
Comment (cat2608) [inline src/snyk/snykCode/views/suggestion/codeSuggestionWebviewScript.ts:11]: Is it correct to say that this package could bring the types we need?

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/53370
https://www.npmjs.com/package/@types/vscode-webview

If so, do you think we could start using them?

File: src/snyk/snykCode/views/suggestion/codeSuggestionWebviewScript.ts [hunk 3]
```diff
@@ -101,7 +104,7 @@
         rule: suggestion.rule,
         id: suggestion.id,
         severity: suggestion.severity,
-        lineOnly: !!lineOnly,
+        lineOnly: lineOnly,
```
Comment (cat2608) [inline src/snyk/snykCode/views/suggestion/codeSuggestionWebviewScript.ts:107]: This operator forces the conversion of truthy/falsy values to primitive booleans. Do we need to work with loose truthy/falsy values?

File: src/snyk/common/constants/views.ts (hunk 1 above)
Comment (cat2608) [reply src/snyk/common/constants/views.ts:16]: I'm not saying to not use a variable to hold the state. I was talking about naming conventions but this is really minor so don't worry about my comment 🙏

**Review by cat2608 — APPROVED** [review]
Thank you :)

File: src/snyk/snykCode/views/suggestion/codeSuggestionWebviewScript.ts (hunk 2 above)
Comment (bastiandoetsch) [reply src/snyk/snykCode/views/suggestion/codeSuggestionWebviewScript.ts:11]: Yes, let's do that, but put that into a different PR

File: src/snyk/snykCode/views/suggestion/codeSuggestionWebviewScript.ts (hunk 3 above)
Comment (bastiandoetsch) [reply src/snyk/snykCode/views/suggestion/codeSuggestionWebviewScript.ts:107]: This seems to be a guaranteed boolean, so we don't need the forced conversion.

**Pull request merged**